STM32F042K4Tx/Core/Src/main.c

```diff
@@ -341,8 +343,17 @@ static void MX_GPIO_Init(void)
 
 /* USER CODE BEGIN 4 */
 void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim) {
+  /* 
+   * tim14 should call interrupt every 200 ms
+   * freq = clock freq / (prescalar + 1) * (counter period + 1) * (repetition counter + 1)
+   *      = 42 MHz/(99 + 1)(41999+1)(1)
+   *      = 42 MHz/(100)(42000)
+   *      = 10 Hz 
+   * period = 1/10 s = 100 ms
+   */ 
+

```

Comment:
Ok, so you are calculating tim14 as your WLoop CAN timer but you are using tim16 for the can init and tim14 for the calculating encoder stuff?

Reply:
Fixed. TIM14 should now just be a counter with no interrupt, and TIM16 now has an interrupt which is called every 500ms.